e2e/docker/test.py

```diff
@@ -0,0 +1,84 @@
+"""Tests for Flower Docker containers."""
+
+import re
+from time import sleep
+from typing import Callable, Union
+
+import pytest
+from testcontainers.compose import ContainerIsNotRunning, DockerCompose
+
+COMPOSE_MANIFEST = "compose.yaml"
+RUN_COMPLETE_STR = "Run finished 3 round(s) in"
+
+
+def test_compose_and_teardown():
+    """Test if Flower containers can successfully start and teardown."""
+    # create the services
+    services = DockerCompose(context=".", compose_file_name=COMPOSE_MANIFEST)
+
+    try:
+        # first the containers do not exist
+        containers = services.get_containers(include_all=True)
+        assert len(containers) == 0
+
+        # then we create them and they exists
+        services.start()
+        containers = services.get_containers(include_all=True)
+        assert len(containers) == 4
+        containers = services.get_containers()
+        assert len(containers) == 4
+
+        # all containers should be running
+        assert containers[0].State == "running"
+        assert containers[1].State == "running"
+        assert containers[2].State == "running"
+        assert containers[3].State == "running"
+
+        # each container has a specific name with the order
+        # determined by the `depends_on` list of services
+        assert containers[0].Service == "serverapp"
+        assert containers[1].Service == "superlink"
+        assert containers[2].Service == "supernode"
+        assert containers[3].Service == "supernode"
+
+        # stop the services but don't remove containers,
+        # networks, volumes, and images
+        services.stop(down=False)
+
+        # Check the state for ServerApp and SuperLink.
+        # TODO: Add test for SuperNode services  # pylint: disable=W0511
```

Comment:
It's a bit disappointing that `get_container` doesn't return an array when you have multiple services with the same `Service` value but a different `Name`.

Reply:
Yes, it's a small drawback. I _could_ [parameterize the test fixture](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html) but I'm planning to revisit this later. IMO, I think these 2 tests are sufficient for now.